src/components/application_manager/src/resumption/resumption_data_db.cc

```diff
@@ -2222,6 +2309,59 @@ bool ResumptionDataDB::InsertChoiceSetData(
   return true;
 }
 
+bool ResumptionDataDB::InsertUserLocationData(
+    const smart_objects::SmartObject& user_location,
+    int64_t application_primary_key) const {
+  SDL_LOG_AUTO_TRACE();
+  using namespace app_mngr;
+  using namespace smart_objects;
+
+  if (user_location.empty()) {
+    SDL_LOG_INFO("Application doesn't contain user location");
+    return true;
+  }
+
+  const auto grid = user_location[rc_rpc_plugin::strings::kGrid];
+  const int32_t col = grid[rc_rpc_plugin::strings::kCol].asInt();
+  const int32_t row = grid[rc_rpc_plugin::strings::kRow].asInt();
+  const int32_t level = grid[rc_rpc_plugin::strings::kLevel].asInt();
+  const int32_t colspan = grid[rc_rpc_plugin::strings::kColspan].asInt();
+  const int32_t rowspan = grid[rc_rpc_plugin::strings::kRowspan].asInt();
+  const int32_t levelspan = grid[rc_rpc_plugin::strings::kLevelspan].asInt();
+
+  utils::dbms::SQLQuery insert_application_user_location(db());
+  if (!insert_application_user_location.Prepare(kInsertUserLocation)) {
+    SDL_LOG_WARN(
+        "Problem with preparation insert "
+        "application user location query");
+    return false;
+  }
+
+  /* Positions of binding data for "insert_application_user_location":
+    field "idApplication" from table "applicationUserLocation" = 0
+    field "col" from table "applicationUserLocation" = 1
+    field "colspan" from table "applicationUserLocation" = 2
+    field "level" from table "applicationUserLocation" = 3
+    field "levelspan" from table "applicationUserLocation" = 4
+    field "row" from table "applicationUserLocation" = 5
+    field "rowspan" from table "applicationUserLocation" = 6*/
+  insert_application_user_location.Bind(0, application_primary_key);
+  insert_application_user_location.Bind(1, col);
+  insert_application_user_location.Bind(2, colspan);
+  insert_application_user_location.Bind(3, level);
+  insert_application_user_location.Bind(4, levelspan);
+  insert_application_user_location.Bind(5, row);
+  insert_application_user_location.Bind(6, rowspan);
+
+  if (!insert_application_user_location.Exec()) {
+    SDL_LOG_WARN("Incorrect insertion of user location");
+    return false;
+  }
+
+  SDL_LOG_INFO("User Location data were saved successfully");
```

Comment:
@VadymLuchko can be removed

Reply:
fixed in bf0ce24216a70351394e14311bafcfac997606f6